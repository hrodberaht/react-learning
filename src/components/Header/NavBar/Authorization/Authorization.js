import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink, withRouter } from 'react-router-dom';

import { logOut } from '../../../../actions';

export class Authorization extends Component {
    rendererLogin = () => {
      const { handleLogOut, auth } = this.props;
      if (auth) return <NavLink to="/login" onClick={handleLogOut}>Log out</NavLink>;

      return <NavLink to="/login">Login</NavLink>;
    }

    render() {
      return (
        this.rendererLogin()
      );
    }
}

const mapStateToProps = state => ({
  auth: state.auth,
});

export default withRouter(connect(mapStateToProps,
  { handleLogOut: logOut })(Authorization));

Authorization.propTypes = {
  auth: PropTypes.bool.isRequired,
  handleLogOut: PropTypes.func.isRequired,
};
