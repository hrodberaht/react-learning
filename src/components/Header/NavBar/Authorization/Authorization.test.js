import React from 'react';
import { shallow } from 'enzyme';
import { Authorization } from './Authorization';

describe('<Authorization />', () => {
  const setup = (propOverrides) => {
    const props = Object.assign({
      auth: false,
      handleLogOut: jest.fn(),
    }, propOverrides);

    const wrapper = shallow(<Authorization {...props} />);

    return {
      props,
      wrapper,
      logOut: wrapper.find('NavLink'),
    };
  };

  it('render', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Log out in nav, if auth is true', () => {
    const wrapper = setup({ auth: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('call handleLogOut after click Log out in nav', () => {
    const { logOut, props: { handleLogOut } } = setup({ auth: true });
    logOut.simulate('click');
    expect(handleLogOut).toHaveBeenCalled();
  });
});
