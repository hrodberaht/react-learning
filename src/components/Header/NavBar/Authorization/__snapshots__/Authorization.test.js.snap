// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Authorization /> render 1`] = `""`;
